crates/bevy_math/src/curve/adaptors.rs

```diff
@@ -0,0 +1,533 @@
+    pub(super) const THIS_CRATE: &str = "bevy_math";
+}
+
+// NOTE ON REFLECTION:
+//
+// Function members of structs pose an obstacle for reflection, because they don't implement
+// reflection traits themselves. Some of these are more problematic than others; for example,
+// `FromReflect` is basically hopeless for function members regardless, so function-containing
+// adaptors will just never be `FromReflect` (at least until function item types implement
+// Default, if that ever happens). Similarly, they do not implement `TypePath`, and as a result,
+// those adaptors also need custom `TypePath` adaptors which use `type_name` instead.
+//
+// The sum total weirdness of the `Reflect` implementations amounts to this; those adaptors:
+// - are currently never `FromReflect`;
+// - have custom `TypePath` implementations which are not fully stable;
+// - have custom `Debug` implementations which display the function only by type name.
+
+/// A curve with a constant value over its domain.
+///
+/// This is a curve that holds an inner value and always produces a clone of that value when sampled.
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
+pub struct ConstantCurve<T> {
+    pub(crate) domain: Interval,
+    pub(crate) value: T,
+}
+
+impl<T> ConstantCurve<T>
+where
+    T: Clone,
+{
+    /// Create a constant curve, which has the given `domain` and always produces the given `value`
+    /// when sampled.
+    pub fn new(domain: Interval, value: T) -> Self {
+        Self { domain, value }
+    }
+}
+
+impl<T> Curve<T> for ConstantCurve<T>
+where
+    T: Clone,
+{
+    #[inline]
+    fn domain(&self) -> Interval {
+        self.domain
+    }
+
+    #[inline]
+    fn sample_unchecked(&self, _t: f32) -> T {
+        self.value.clone()
+    }
+}
+
+/// A curve defined by a function together with a fixed domain.
+///
+/// This is a curve that holds an inner function `f` which takes numbers (`f32`) as input and produces
+/// output of type `T`. The value of this curve when sampled at time `t` is just `f(t)`.
+#[derive(Clone)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "bevy_reflect",
+    derive(Reflect),
+    reflect(where T: TypePath),
+    reflect(from_reflect = false, type_path = false),
+)]
+pub struct FunctionCurve<T, F> {
+    pub(crate) domain: Interval,
+    #[cfg_attr(feature = "bevy_reflect", reflect(ignore))]
+    pub(crate) f: F,
+    #[cfg_attr(feature = "bevy_reflect", reflect(ignore))]
+    pub(crate) _phantom: PhantomData<T>,
+}
+
+impl<T, F> Debug for FunctionCurve<T, F> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("FunctionCurve")
+            .field("domain", &self.domain)
+            .field("f", &type_name::<F>())
+            .field("_phantom", &self._phantom)
```

Comment:
Do you need to print the phantom data here?

Reply:
Probably not! Removed.